api/types/events.go

```diff
@@ -147,6 +155,38 @@ func (kind WatchKind) IsTrivial() bool {
 	return kind.SubKind == "" && kind.Name == "" && kind.Version == "" && !kind.LoadSecrets && len(kind.Filter) == 0
 }
 
+// IsSupersetOf determines whether kind (receiver) targets exactly the same or a wider scope of events
+// as the given subset kind.
+// Generally this means that if kind specifies a filter, its subset must have exactly the same or a narrower one.
+func (kind WatchKind) IsSupersetOf(subset WatchKind) bool {
+	// kind and subkind must always be equal
+	if kind.Kind != subset.Kind || kind.SubKind != subset.SubKind {
+		return false
+	}
+
+	// exception to the general rule: we don't fail in the situation where `kind.Version != "" && subset.Version == ""`
+	// because we assume that subset is version-agnostic and will be satisfied with the version filtered by the superset.
+	if kind.Version != "" && subset.Version != "" && kind.Version != subset.Version {
```

Comment:
The only resource that utilizes `Version` field is `KindAppServer` and the comment next to it says ["delete in 9.0"](https://github.com/gravitational/teleport/blob/2f09551c7cbb95e40b3664ef89718fa6fa4597c7/lib/services/local/events.go#L108).

So, alternatively, we can clean that up and ignore resource versions everywhere with regards to event streams.

Reply:
Yeah we can clean it up now.

Reply:
That's great, I was able to clean up `newAppServerV2Parser` along with all event versioning logic and the extra `kindSubkind` type: https://github.com/gravitational/teleport/pull/24080/commits/34fa740ae2f56eaff897ac0776cfedb1b97ec8b3